automap/src/comm_layer/pcp.rs

```diff
@@ -0,0 +1,2203 @@
+        init_test_logging();
+        let (tx, rx) = unbounded();
+        let mut announce_packet = vanilla_response();
+        announce_packet.opcode = Opcode::Announce;
+        announce_packet.lifetime = 0;
+        let mut announce_buf = [0u8; 100];
+        let announce_packet_len = announce_packet.marshal(&mut announce_buf).unwrap();
+        let socket: Box<dyn UdpSocketWrapper> = Box::new(
+            UdpSocketMock::new()
+                .set_read_timeout_result(Ok(()))
+                .recv_from_result(
+                    Ok((
+                        announce_packet_len,
+                        SocketAddr::from_str("1.1.1.1:1111").unwrap(),
+                    )),
+                    announce_buf.to_vec(),
+                ),
+        );
+        let change_log_arc = Arc::new(Mutex::new(vec![]));
+        let inner_cla = change_log_arc.clone();
+        let change_handler = Box::new(move |change| {
+            let mut change_log = inner_cla.lock().unwrap();
+            change_log.push(change)
+        });
+        let mapping_transactor = Box::new(
+            MappingTransactorMock::new()
+                .mapping_transaction_result(Ok((1111, *vanilla_map_response()))),
+        );
+        let logger = Logger::new(
+            "thread_guts_logs_and_continues_if_announcement_is_received_before_mapping_config",
+        );
+
+        let handle = thread::spawn(move || {
+            let _ = PcpTransactor::thread_guts(
+                socket.as_ref(),
+                &rx,
+                Arc::new(Mutex::new(PcpTransactorInner {
+                    mapping_transactor,
+                    factories: Factories::default(),
+                })),
+                SocketAddr::new(IpAddr::from_str("1.1.1.1").unwrap(), 0),
+                change_handler,
+                10,
+                logger,
+            );
+        });
+
+        thread::sleep(Duration::from_millis(100));
+        tx.send(HousekeepingThreadCommand::Stop).unwrap();
+        handle.join().unwrap();
+        let change_log = change_log_arc.lock().unwrap();
+        assert_eq!(
+            *change_log,
+            vec![AutomapChange::NewIp(
+                IpAddr::from_str("72.73.74.75").unwrap()
+            )]
+        );
+        TestLogHandler::new ().exists_log_containing(
+            "DEBUG: thread_guts_logs_and_continues_if_announcement_is_received_before_mapping_config: Received announcement that public IP address changed to 72.73.74.75"
+        );
+    }
+
+    #[test]
+    fn thread_guts_complains_if_remapping_fails() {
+        init_test_logging();
+        let socket_addr = SocketAddr::from_str("1.1.1.1:1").unwrap();
+        let (tx, rx) = unbounded();
+        let socket: Box<dyn UdpSocketWrapper> = Box::new(
+            UdpSocketMock::new()
+                .set_read_timeout_result(Ok(()))
+                .recv_from_result(Ok((5, socket_addr)), b"booga".to_vec()),
```

Comment:
I'm running out of time but I don't understand why these values are used here as the result when the test should be focused on a different error. One test which is situated higher uses this result to simulate an unparsable Pcp packet.

"thread_guts_logs_if_unparseable_pcp_packet_arrives"